src/components/TradeWidget/index.tsx

```diff
@@ -62,26 +76,33 @@ const TradeWidget: React.FC = () => {
 
   return (
     <WrappedWidget>
-      <TokenRow
-        token={sellToken}
-        tokens={tokens}
-        balance={sellTokenBalance}
-        selectLabel="sell"
-        onSelectChange={onSelectChangeFactory(setSellToken, receiveToken)}
-      />
-      <IconWrapper onClick={swapTokens}>
-        <FontAwesomeIcon icon={faExchangeAlt} rotation={90} size="2x" />
-      </IconWrapper>
-      <TokenRow
-        token={receiveToken}
-        tokens={tokens}
-        balance={receiveTokenBalance}
-        selectLabel="receive"
-        onSelectChange={onSelectChangeFactory(setReceiveToken, sellToken)}
-      />
-      <SubmitButton>
-        <FontAwesomeIcon icon={faPaperPlane} size="lg" /> Send limit order
-      </SubmitButton>
+      <FormContext {...methods}>
+        <WrapperForm onSubmit={methods.handleSubmit(data => console.log('data', data))}>
+          <TokenRow
+            token={sellToken}
+            tokens={tokens}
+            balance={sellTokenBalance}
+            selectLabel="sell"
+            onSelectChange={onSelectChangeFactory(setSellToken, receiveToken)}
```

Comment:
My question is not not from this PR I guess, but why the word `Factory` here?

Reply:
That's the metaphor I could thing of a function that returns a function. Is there anything better?